### PR TITLE
Rerender, and fix `libgoogle-cloud` issues

### PR DESCRIPTION
**Pull request by johnkerl**
This is on `main` with tiledbsoma 1.7.2. See also #91 which attempts a pre-check for 1.8.0.

This doesn't bump any application-level versions -- hence build number 1. This works around problems that have recently arisen in our dependencies, problems which made a previously solvable recipe no longer solvable.

Solver issue:

```
744:    Encountered problems while solving:
745-      - package tiledb-2.19.1-hc0b898c_0 requires libcurl >=7.87.0,<8.0a0, but none of the providers can be installed
746-
747-    Could not solve for environment specs
748-    The following packages are incompatible
749-    ├─ libgoogle-cloud-storage 2.21.*  is installable and it requires
750-    │  └─ libgoogle-cloud [2.21.0 h31df0ca_0|2.21.0 h31df0ca_1|2.21.0 h72bcb37_2], which requires
751-    │     └─ libcurl >=8.5.0,<9.0a0 , which can be installed;
752-    └─ tiledb >=2.19.1,<2.20  is not installable because there are no viable options
753-       ├─ tiledb 2.19.1 would require
754-       │  └─ libgoogle-cloud >=2.12.0,<2.13.0a0 , which conflicts with any installable versions previously reported;
755-       ├─ tiledb 2.19.1 would require
756-       │  └─ libcurl >=7.88.1,<8.0a0 , which conflicts with any installable versions previously reported;
757-       └─ tiledb 2.19.1 would require
758-          └─ libcurl >=7.87.0,<8.0a0 , which conflicts with any installable versions previously reported.
```

```
tiledb 2.19.1 requires libgoogle-cloud 2.12.*
```

Supporting info:

* `google-cloud-cpp`
  * https://anaconda.org/conda-forge/google-cloud-cpp/files
  * Has 2.11 2.12 2.21
* `libgoogle-cloud`
  * https://anaconda.org/conda-forge/libgoogle-cloud
  * Has 2.11 2.12 2.17 2.21
* `libgoogle-cloud-storage`
  * https://anaconda.org/conda-forge/libgoogle-cloud-storage
  * Has 2.17 2.21

See also:

<details>

https://anaconda.org/conda-forge/tiledb/files?version=2.19.1
```
Type  Size  Name
conda 5.8MB linux-64/tiledb-2.19.1-h4386cac_0.conda
conda 5.8MB linux-64/tiledb-2.19.1-h584b59b_0.conda
conda 5.8MB linux-64/tiledb-2.19.1-hc0b898c_0.conda
```

https://anaconda.org/conda-forge/tiledb/2.19.1/download/linux-64/tiledb-2.19.1-h4386cac_0.conda

```
wget https://anaconda.org/conda-forge/tiledb/2.19.1/download/linux-64/tiledb-2.19.1-h4386cac_0.conda
```

```
ubuntu@anzo[prod][][~]$ conda inspect hash-inputs tiledb-2.19.1-h4386cac_0.conda
{'tiledb-2.19.1-h4386cac_0.conda': {'recipe': {'bzip2': '1',
                                               'channel_targets': 'conda-forge '
                                                                  'main',
                                               'cxx_compiler': 'gxx',
                                               'cxx_compiler_version': '12',
                                               'gcs': 'gcs_enabled',
                                               'google_cloud_cpp': '2.12',
                                               'libabseil': '20230802',
                                               'libcurl': '8',
                                               'libxml2': '2',
                                               'lz4_c': '1.9.3',
                                               'openssl': '3',
                                               'target_platform': 'linux-64',
                                               'zlib': '1.2',
                                               'zstd': '1.5'}}}
```

https://anaconda.org/tiledb/tiledb/files?channel=for-cloud&version=2.19.1

```
wget https://anaconda.org/tiledb/tiledb/2.19.1/download/linux-64/tiledb-2.19.1-h24029db_1.conda
```

```
conda inspect hash-inputs tiledb-2.19.1-h24029db_1.conda
{'tiledb-2.19.1-h24029db': {'recipe': {'channel_targets': 'tiledb for-cloud',
                                       'cxx_compiler': 'gxx',
                                       'cxx_compiler_version': '12',
                                       'target_platform': 'linux-64'}}}
```

https://github.com/conda-forge/tiledb-feedstock/blob/295eacda3ca366b2795b526c5de233931e5aa2a2/recipe/meta.yaml

</details>

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -100,15 +102,15 @@ outputs:
...
+        - anndata       # [py>37]
         - anndata <0.9  # [py<=37]
         - tiledb-py >=0.25.0,<0.26.0
         - typing-extensions >=4.1
         - numba >=0.58.1 # [py>37]
         - numba          # [py<=37]
         - attrs >=22.2
         - somacore >=1.0.7
-        - scanpy 1.9.*
+        - scanpy >=1.9.2
```
Comment (johnkerl) [inline recipe/meta.yaml:113]: Matching `tiledbsoma`'s `apis/python/setup.py`

File: recipe/meta.yaml [hunk 2]
```diff
@@ -100,15 +102,15 @@ outputs:
         - pyarrow-hotfix
         - python
         - scipy
-        - anndata <0.10 # [py>37]
+        - anndata       # [py>37]
```
Comment (johnkerl) [inline recipe/meta.yaml:105]: Matching `tiledbsoma`'s `apis/python/setup.py`

**Comment by johnkerl** [comment]
[sc-42198]

**Comment by johnkerl** [comment]
Here's what happens with `libgoogle-cloud` 2.21 rather than 2.12:

https://gist.github.com/johnkerl/7aadef919c83844ee48728787e791e6d

**Review by jdblischak — APPROVED** [review]
I'm fine with this libgoogle-cloud workaround until we can identify a better long-term solution

**Pull request merged**